syntax = "proto3";

package nt_msg;

message Element {
  enum MsgType {
    MSG_TYPE_ERROR = 0;
    MSG_TYPE_TEXT = 1;
    MSG_TYPE_PIC = 2;
  }

  int64 elementId = 45001;
  int32 elementType = 45002;

  // TextElement
  optional string textStr = 45101;
  optional int32 atType = 45102;

  // PicElement
  optional string fileName = 45402;
  optional int32 fileSize = 45405;
  optional string unknown_45406 = 45406;
  optional int32 picWidth = 45411;
  optional int32 picHeight = 45412;
  optional int32 unknown_45413 = 45413;
  optional int32 unknown_45414 = 45414;
  optional int32 picType = 45416;
  optional int32 unknown_45418 = 45418;
  optional string fileUuid = 45503;
  optional string imageUrl198 = 45802;
  optional string imageUrl720 = 45803;
  optional string originImageUrl = 45804;
  optional int32 unknown_45805 = 45805;
  optional int32 unknown_45806 = 45806;
  optional int32 unknown_45807 = 45807;
  optional int32 unknown_45829 = 45829;

  // ReplyElement
  message SourceMsgTextElem {
    int64 unknown_45001 = 45001;
    int32 replyAbsElemType = 45002;
    string textElemContent = 45101;
  }
  optional string senderUidStr = 40020;
  optional int32 replayMsgSeq = 47402;
  optional int32 senderUid = 47403;
  optional int32 replyMsgTime = 47404;
  optional int32 unknown_47411 = 47411;
  optional int64 unknown_47416 = 47416;
  optional int64 unknown_47422 = 47422;
  repeated SourceMsgTextElem sourceMsgTextElems = 47423;


}

message Elements {
  repeated Element elem = 40800;
}